source/extensions/filters/common/rbac/utility.h

```diff
@@ -38,21 +38,43 @@ RoleBasedAccessControlFilterStats
 generateStats(const std::string& prefix, const std::string& shadow_prefix, Stats::Scope& scope);
 
 template <class ConfigType>
-std::unique_ptr<RoleBasedAccessControlEngineImpl>
-createEngine(const ConfigType& config, ProtobufMessage::ValidationVisitor& validation_visitor) {
-  return config.has_rules() ? std::make_unique<RoleBasedAccessControlEngineImpl>(
-                                  config.rules(), validation_visitor, EnforcementMode::Enforced)
-                            : nullptr;
+std::unique_ptr<RoleBasedAccessControlEngine>
+createEngine(const ConfigType& config, Server::Configuration::ServerFactoryContext& context,
+             ProtobufMessage::ValidationVisitor& validation_visitor,
+             ActionValidationVisitor& action_validation_visitor) {
+  if (config.has_matcher()) {
+    if (config.has_rules()) {
+      ENVOY_LOG_MISC(warn, "RBAC rules are ignored");
```

Comment:
nit, better to also explain why it's ignored, e.g. `"RBAC rules are ignored when matcher is configured"`

Reply:
Done.